File: banda/AMarotoChivite-banda/AMarotoChivite-banda-Java/src/main/java/simulacionBandaMusica/factories/CantanteProFactory.java
package simulacionBandaMusica.factories;

import simulacionBandaMusica.enums.TipoGuitarra;
import simulacionBandaMusica.models.CantantePro;

import static simulacionBandaMusica.utils.sorteoDatos.*;

public class CantanteProFactory {

    public static CantantePro create() {
        String name = randomName();
        int experienceYear = randomAnno();
        int capacidadPulmonar = randomPulmonarCapacity();
        TipoGuitarra tipoGuitarra = randomTipoGuitarra();

        return new CantantePro(name, "Voz y guitarra", experienceYear, capacidadPulmonar, tipoGuitarra);
    }
}

File: banda/AMarotoChivite-banda/AMarotoChivite-banda-Kotlin/src/main/kotlin/simulacionBandaMusica/factories/CantanteProFactory.kt
package simulacionBandaMusica.factories

import simulacionBandaMusica.models.Musico
import simulacionBandaMusica.utils.sorteoDatos

object CantanteProFactory {
    fun create(): Musico.CantantePro {
        val name = sorteoDatos.randomName()
        val experienceYear = sorteoDatos.randomAnno()
        val capacidadPulmonar = sorteoDatos.randomPulmonarCapacity()
        val tipoGuitarra = sorteoDatos.randomTipoGuitarra()
        return Musico.CantantePro(name, "Voz y guitarra", experienceYear, capacidadPulmonar, tipoGuitarra)
    }
}